### PR TITLE
refactor(base): remove all calls to repo.CanonicalizeDatasetRef from base package

### PR DESCRIPTION
**Pull request by b5**
makes progress on https://github.com/qri-io/qri/issues/1437 building on save path refactoring to move the `base` package away from `repo.CanonicalizeDatasetRef`

### CODE REVIEW

File: cmd/save.go [hunk 1]
```diff
@@ -184,19 +185,21 @@ continue?`, true) {
...
 	if err = o.DatasetMethods.Save(p, res); err != nil {
 		return err
 	}
 
 	o.StopSpinner()
-	printSuccess(o.ErrOut, "dataset saved: %s", res)
-	if res.Dataset.Structure != nil && res.Dataset.Structure.ErrCount > 0 {
-		printWarning(o.ErrOut, fmt.Sprintf("this dataset has %d validation errors", res.Dataset.Structure.ErrCount))
+	ref := dsref.ConvertDatasetToVersionInfo(res).SimpleRef()
+	ref.ProfileID = ""
```
Comment (Arqu) [inline cmd/save.go:195]: Why do we remove the profileID here?

File: remote/client.go [hunk 2]
```diff
@@ -453,73 +452,15 @@ func (c *client) PullDataset(ctx context.Context, ref *dsref.Ref, remoteAddr str
...
-	success := false
-	for i := 0; i < tasks; i++ {
-		res := <-responses
-		err = res.Error
-		if err == nil {
-			cancelFetch()
-			success = true
-			*ref = *res.Ref
-			break
-		}
```
Comment (Arqu) [inline remote/client.go:517]: 🤩

File: base/dataset.go [hunk 3]
```diff
@@ -260,45 +260,6 @@ func RawDatasetRefs(ctx context.Context, r repo.Repo) (string, error) {
 	return builder.String(), nil
 }
 
-// FetchDataset grabs a dataset from a remote source
-func FetchDataset(ctx context.Context, r repo.Repo, ref *reporef.DatasetRef, pin, load bool) (err error) {
```
Comment (Arqu) [inline base/dataset.go:264]: What replaces FetchDataset, didn't seem to find the the alternative?

Comment (b5) [reply base/dataset.go:264]: Yeah there _isn't_ an alternative right now. FetchDataset doesn't currently fetch logbook data, and this logic really needs to be moved into the `p2p` package. I'd like to follow up in the very near future with a PR that replaces `FetchDataset`. Will open an issue

File: cmd/save.go (hunk 1 above)
Comment (b5) [reply cmd/save.go:195]: to make the call to `ref.String()` below not print such a verbose reference.

without profileID:
```
dataset saved rm_me_3/saxaphones@/ipfs/QmRL5Wq4K1JXYZsgZKWJpdFJvQEsNCc6n9twRkzSGsCLbF
```

with:
```
dataset saved  rm_me_3/saxaphones@QmcPmrQYqFamdHYFL8qbfpbT3dRNRVanLmFEY9srWsddt9/ipfs/QmRL5Wq4K1JXYZsgZKWJpdFJvQEsNCc6n9twRkzSGsCLbF
```

Long term it'd be better to add a method to `dsref` that does this

**Pull request merged**